### ISSUE TITLE
[ ERROR ] Bench Start

### ISSUE THREAD

**Issue by giuseppelamatrice** [frappe/erpnext#13787]
```
~/frappe-bench$ bench start
10:21:57 system           | redis_socketio.1 started (pid=11693)
10:21:57 system           | redis_queue.1 started (pid=11704)
10:21:57 system           | watch.1 started (pid=11697)
10:21:57 redis_socketio.1 | 11701:M 25 Apr 10:21:57.412 * Increased maximum number of open files to 10032 (it was originally set to 1024).
10:21:57 redis_socketio.1 |                 _._                                                  
10:21:57 redis_socketio.1 |            _.-``__ ''-._                                             
10:21:57 redis_socketio.1 |       _.-``    `.  `_.  ''-._           Redis 3.0.6 (00000000/0) 64 bit
10:21:57 redis_socketio.1 |   .-`` .-```.  ```\/    _.,_ ''-._                                   
10:21:57 redis_socketio.1 |  (    '      ,       .-`  | `,    )     Running in standalone mode
10:21:57 redis_socketio.1 |  |`-._`-...-` __...-.``-._|'` _.-'|     Port: 12000
10:21:57 redis_socketio.1 |  |    `-._   `._    /     _.-'    |     PID: 11701
10:21:57 redis_socketio.1 |   `-._    `-._  `-./  _.-'    _.-'                                   
10:21:57 redis_socketio.1 |  |`-._`-._    `-.__.-'    _.-'_.-'|                                  
10:21:57 redis_socketio.1 |  |    `-._`-._        _.-'_.-'    |           http://redis.io        
10:21:57 redis_socketio.1 |   `-._    `-._`-.__.-'_.-'    _.-'                                   
10:21:57 redis_socketio.1 |  |`-._`-._    `-.__.-'    _.-'_.-'|                                  
10:21:57 redis_socketio.1 |  |    `-._`-._        _.-'_.-'    |                                  
10:21:57 redis_socketio.1 |   `-._    `-._`-.__.-'_.-'    _.-'                                   
10:21:57 redis_socketio.1 |       `-._    `-.__.-'    _.-'                                       
10:21:57 redis_socketio.1 |           `-._        _.-'                                           
10:21:57 redis_socketio.1 |               `-.__.-'                                               
10:21:57 redis_socketio.1 | 
10:21:57 redis_socketio.1 | 11701:M 25 Apr 10:21:57.420 # WARNING: The TCP backlog setting of 511 cannot be enforced because /proc/sys/net/core/somaxconn is set to the lower value of 128.
10:21:57 redis_socketio.1 | 11701:M 25 Apr 10:21:57.422 # Server started, Redis version 3.0.6
10:21:57 redis_socketio.1 | 11701:M 25 Apr 10:21:57.422 # WARNING overcommit_memory is set to 0! Background save may fail under low memory condition. To fix this issue add 'vm.overcommit_memory = 1' to /etc/sysctl.conf and then reboot or run the command 'sysctl vm.overcommit_memory=1' for this to take effect.
10:21:57 redis_socketio.1 | 11701:M 25 Apr 10:21:57.422 # WARNING you have Transparent Huge Pages (THP) support enabled in your kernel. This will create latency and memory usage issues with Redis. To fix this issue run the command 'echo never > /sys/kernel/mm/transparent_hugepage/enabled' as root, and add it to your /etc/rc.local in order to retain the setting after a reboot. Redis must be restarted after THP is disabled.
10:21:57 redis_socketio.1 | 11701:M 25 Apr 10:21:57.422 * The server is now ready to accept connections on port 12000
10:21:57 system           | schedule.1 started (pid=11711)
10:21:57 system           | redis_cache.1 started (pid=11710)
10:21:57 redis_cache.1    | 11716:M 25 Apr 10:21:57.421 * Increased maximum number of open files to 10032 (it was originally set to 1024).
10:21:57 redis_cache.1    |                 _._                                                  
10:21:57 redis_cache.1    |            _.-``__ ''-._                                             
10:21:57 redis_cache.1    |       _.-``    `.  `_.  ''-._           Redis 3.0.6 (00000000/0) 64 bit
10:21:57 redis_cache.1    |   .-`` .-```.  ```\/    _.,_ ''-._                                   
10:21:57 redis_cache.1    |  (    '      ,       .-`  | `,    )     Running in standalone mode
10:21:57 redis_cache.1    |  |`-._`-...-` __...-.``-._|'` _.-'|     Port: 13000
10:21:57 redis_cache.1    |  |    `-._   `._    /     _.-'    |     PID: 11716
10:21:57 redis_cache.1    |   `-._    `-._  `-./  _.-'    _.-'                                   
10:21:57 redis_cache.1    |  |`-._`-._    `-.__.-'    _.-'_.-'|                                  
10:21:57 redis_cache.1    |  |    `-._`-._        _.-'_.-'    |           http://redis.io        
10:21:57 redis_cache.1    |   `-._    `-._`-.__.-'_.-'    _.-'                                   
10:21:57 redis_cache.1    |  |`-._`-._    `-.__.-'    _.-'_.-'|                                  
10:21:57 redis_cache.1    |  |    `-._`-._        _.-'_.-'    |                                  
10:21:57 redis_cache.1    |   `-._    `-._`-.__.-'_.-'    _.-'                                   
10:21:57 redis_cache.1    |       `-._    `-.__.-'    _.-'                                       
10:21:57 redis_cache.1    |           `-._        _.-'                                           
10:21:57 redis_cache.1    |               `-.__.-'                                               
10:21:57 redis_cache.1    | 
10:21:57 redis_cache.1    | 11716:M 25 Apr 10:21:57.422 # WARNING: The TCP backlog setting of 511 cannot be enforced because /proc/sys/net/core/somaxconn is set to the lower value of 128.
10:21:57 redis_cache.1    | 11716:M 25 Apr 10:21:57.422 # Server started, Redis version 3.0.6
10:21:57 system           | socketio.1 started (pid=11708)
10:21:57 system           | worker_short.1 started (pid=11735)
10:21:57 system           | web.1 started (pid=11733)
10:21:57 redis_queue.1    | 11706:M 25 Apr 10:21:57.410 * Increased maximum number of open files to 10032 (it was originally set to 1024).
10:21:57 redis_queue.1    |                 _._                                                  
10:21:57 redis_queue.1    |            _.-``__ ''-._                                             
10:21:57 redis_queue.1    |       _.-``    `.  `_.  ''-._           Redis 3.0.6 (00000000/0) 64 bit
10:21:57 redis_queue.1    |   .-`` .-```.  ```\/    _.,_ ''-._                                   
10:21:57 redis_queue.1    |  (    '      ,       .-`  | `,    )     Running in standalone mode
10:21:57 redis_queue.1    |  |`-._`-...-` __...-.``-._|'` _.-'|     Port: 11000
10:21:57 redis_queue.1    |  |    `-._   `._    /     _.-'    |     PID: 11706
10:21:57 redis_queue.1    |   `-._    `-._  `-./  _.-'    _.-'                                   
10:21:57 redis_queue.1    |  |`-._`-._    `-.__.-'    _.-'_.-'|                                  
10:21:57 redis_queue.1    |  |    `-._`-._        _.-'_.-'    |           http://redis.io        
10:21:57 redis_queue.1    |   `-._    `-._`-.__.-'_.-'    _.-'                                   
10:21:57 redis_queue.1    |  |`-._`-._    `-.__.-'    _.-'_.-'|                                  
10:21:57 redis_queue.1    |  |    `-._`-._        _.-'_.-'    |                                  
10:21:57 redis_queue.1    |   `-._    `-._`-.__.-'_.-'    _.-'                                   
10:21:57 redis_queue.1    |       `-._    `-.__.-'    _.-'                                       
10:21:57 redis_queue.1    |           `-._        _.-'                                           
10:21:57 redis_queue.1    |               `-.__.-'                                               
10:21:57 redis_queue.1    | 
10:21:57 redis_queue.1    | 11706:M 25 Apr 10:21:57.415 # WARNING: The TCP backlog setting of 511 cannot be enforced because /proc/sys/net/core/somaxconn is set to the lower value of 128.
10:21:57 redis_queue.1    | 11706:M 25 Apr 10:21:57.415 # Server started, Redis version 3.0.6
10:21:57 redis_queue.1    | 11706:M 25 Apr 10:21:57.415 # WARNING overcommit_memory is set to 0! Background save may fail under low memory condition. To fix this issue add 'vm.overcommit_memory = 1' to /etc/sysctl.conf and then reboot or run the command 'sysctl vm.overcommit_memory=1' for this to take effect.
10:21:57 redis_queue.1    | 11706:M 25 Apr 10:21:57.415 # WARNING you have Transparent Huge Pages (THP) support enabled in your kernel. This will create latency and memory usage issues with Redis. To fix this issue run the command 'echo never > /sys/kernel/mm/transparent_hugepage/enabled' as root, and add it to your /etc/rc.local in order to retain the setting after a reboot. Redis must be restarted after THP is disabled.
10:21:57 redis_queue.1    | 11706:M 25 Apr 10:21:57.422 * The server is now ready to accept connections on port 11000
10:21:57 system           | worker_long.1 started (pid=11695)
10:21:57 system           | worker_default.1 started (pid=11709)
10:21:57 redis_cache.1    | 11716:M 25 Apr 10:21:57.422 # WARNING overcommit_memory is set to 0! Background save may fail under low memory condition. To fix this issue add 'vm.overcommit_memory = 1' to /etc/sysctl.conf and then reboot or run the command 'sysctl vm.overcommit_memory=1' for this to take effect.
10:21:57 redis_cache.1    | 11716:M 25 Apr 10:21:57.422 # WARNING you have Transparent Huge Pages (THP) support enabled in your kernel. This will create latency and memory usage issues with Redis. To fix this issue run the command 'echo never > /sys/kernel/mm/transparent_hugepage/enabled' as root, and add it to your /etc/rc.local in order to retain the setting after a reboot. Redis must be restarted after THP is disabled.
10:21:57 redis_cache.1    | 11716:M 25 Apr 10:21:57.422 * The server is now ready to accept connections on port 13000
10:21:59 socketio.1       | listening on *: 9000
10:22:02 web.1            |  * Running on http://0.0.0.0:8000/ (Press CTRL+C to quit)
10:22:02 web.1            |  * Restarting with inotify reloader
10:22:03 watch.1          | 
10:22:03 watch.1          | > frappe@ watch /home/giuseppe/frappe-bench/apps/frappe
10:22:03 watch.1          | > node rollup/watch.js
10:22:03 watch.1          | 
10:22:03 web.1            |  * Debugger is active!
10:22:03 web.1            |  * Debugger PIN: 219-997-878
10:22:04 watch.1          | 
10:22:04 watch.1          | Rollup Watcher Started
10:22:04 watch.1          | 
10:22:04 watch.1          | Watching...
10:22:04 watch.1          | Error in: undefined
10:22:04 watch.1          | Error: watch /home/giuseppe/frappe-bench/apps/frappe/frappe/public/css/font-awesome.css ENOSPC
10:22:05 system           | watch.1 stopped (rc=0)
10:22:05 system           | sending SIGTERM to redis_socketio.1 (pid 11693)
10:22:05 system           | sending SIGTERM to worker_long.1 (pid 11695)
10:22:05 system           | sending SIGTERM to redis_queue.1 (pid 11704)
10:22:05 system           | sending SIGTERM to web.1 (pid 11733)
10:22:05 system           | sending SIGTERM to schedule.1 (pid 11711)
10:22:05 system           | sending SIGTERM to worker_default.1 (pid 11709)
10:22:05 system           | sending SIGTERM to socketio.1 (pid 11708)
10:22:05 system           | sending SIGTERM to redis_cache.1 (pid 11710)
10:22:05 system           | sending SIGTERM to worker_short.1 (pid 11735)
10:22:05 redis_socketio.1 | 11701:signal-handler (1524644525) Received SIGTERM scheduling shutdown...
10:22:05 worker_short.1   | 10:22:05 Warm shut down requested
10:22:05 worker_long.1    | 10:22:05 Warm shut down requested
10:22:05 redis_cache.1    | 11716:signal-handler (1524644525) Received SIGTERM scheduling shutdown...
10:22:05 redis_cache.1    | 11716:M 25 Apr 10:22:05.037 # User requested shutdown...
10:22:05 system           | schedule.1 stopped (rc=-15)
10:22:05 redis_cache.1    | 11716:M 25 Apr 10:22:05.037 # Redis is now ready to exit, bye bye...
10:22:05 system           | socketio.1 stopped (rc=-15)
10:22:05 worker_default.1 | 10:22:05 Warm shut down requested
10:22:05 redis_queue.1    | 11706:signal-handler (1524644525) Received SIGTERM scheduling shutdown...
10:22:05 system           | redis_cache.1 stopped (rc=-15)
10:22:05 redis_queue.1    | 11706:M 25 Apr 10:22:05.125 # User requested shutdown...
10:22:05 redis_queue.1    | 11706:M 25 Apr 10:22:05.125 # Redis is now ready to exit, bye bye...
10:22:05 system           | redis_queue.1 stopped (rc=-15)
10:22:05 redis_socketio.1 | 11701:M 25 Apr 10:22:05.129 # User requested shutdown...
10:22:05 redis_socketio.1 | 11701:M 25 Apr 10:22:05.129 # Redis is now ready to exit, bye bye...
10:22:05 system           | worker_default.1 stopped (rc=-15)
10:22:05 system           | worker_long.1 stopped (rc=-15)
10:22:05 system           | redis_socketio.1 stopped (rc=-15)
10:22:05 system           | worker_short.1 stopped (rc=-15)
10:22:05 system           | web.1 stopped (rc=-15)

```

**Comment by umairsy** [comment]
Hi,

Issue validated on the discuss forum?

@codingCoffee can you please review?

**Comment by codingCoffee** [comment]
It's an ENOSPC error.
TL;DR
```
echo fs.inotify.max_user_watches=524288 | sudo tee -a /etc/sysctl.conf && sudo sysctl -p
```
Ref: https://stackoverflow.com/questions/22475849/node-js-error-enospc


You may refer to the following to know more about it:
https://discuss.erpnext.com/t/bench-fails-to-start-after-a-bench-update/35045/16

It is not something we can fix inside ERPNext or Frappe framework for that mater.

**Comment by giuseppelamatrice** [comment]
This is my log:

`Exception happened during processing of request from ('127.0.0.1', 48618)
16:39:01 web.1            | Traceback (most recent call last):
16:39:01 web.1            |   File "/usr/lib/python2.7/SocketServer.py", line 596, in process_request_thread
16:39:01 web.1            |     self.finish_request(request, client_address)
16:39:01 web.1            |   File "/usr/lib/python2.7/SocketServer.py", line 331, in finish_request
16:39:01 web.1            |     self.RequestHandlerClass(request, client_address, self)
16:39:01 web.1            |   File "/usr/lib/python2.7/SocketServer.py", line 652, in __init__
16:39:01 web.1            |     self.handle()
16:39:01 web.1            |   File "/home/giuseppe/frappe-bench/env/local/lib/python2.7/site-packages/werkzeug/serving.py", line 293, in handle
16:39:01 web.1            |     rv = BaseHTTPRequestHandler.handle(self)
16:39:01 web.1            |   File "/usr/lib/python2.7/BaseHTTPServer.py", line 340, in handle
16:39:01 web.1            |     self.handle_one_request()
16:39:01 web.1            |   File "/home/giuseppe/frappe-bench/env/local/lib/python2.7/site-packages/werkzeug/serving.py", line 328, in handle_one_request
16:39:01 web.1            |     return self.run_wsgi()
16:39:01 web.1            |   File "/home/giuseppe/frappe-bench/env/local/lib/python2.7/site-packages/werkzeug/serving.py", line 270, in run_wsgi
16:39:01 web.1            |     execute(self.server.app)
16:39:01 web.1            |   File "/home/giuseppe/frappe-bench/env/local/lib/python2.7/site-packages/werkzeug/serving.py", line 260, in execute
16:39:01 web.1            |     for data in application_iter:
16:39:01 web.1            |   File "/home/giuseppe/frappe-bench/env/local/lib/python2.7/site-packages/werkzeug/debug/__init__.py", line 288, in debug_application
16:39:01 web.1            |     app_iter = self.app(environ, start_response)
16:39:01 web.1            |   File "/home/giuseppe/frappe-bench/apps/frappe/frappe/middlewares.py", line 15, in __call__
16:39:01 web.1            |     return super(StaticDataMiddleware, self).__call__(environ, start_response)
16:39:01 web.1            |   File "/home/giuseppe/frappe-bench/env/local/lib/python2.7/site-packages/werkzeug/wsgi.py", line 766, in __call__
16:39:01 web.1            |     return self.app(environ, start_response)
16:39:01 web.1            |   File "/home/giuseppe/frappe-bench/env/local/lib/python2.7/site-packages/werkzeug/wsgi.py", line 766, in __call__
16:39:01 web.1            |     return self.app(environ, start_response)
16:39:01 web.1            |   File "/home/giuseppe/frappe-bench/env/local/lib/python2.7/site-packages/werkzeug/local.py", line 228, in application
16:39:01 web.1            |     return ClosingIterator(app(environ, start_response), self.cleanup)
16:39:01 web.1            |   File "/home/giuseppe/frappe-bench/env/local/lib/python2.7/site-packages/werkzeug/wrappers.py", line 308, in application
16:39:01 web.1            |     resp = f(*args[:-2] + (request,))
16:39:01 web.1            |   File "/home/giuseppe/frappe-bench/apps/frappe/frappe/app.py", line 59, in application
16:39:01 web.1            |     init_request(request)
16:39:01 web.1            |   File "/home/giuseppe/frappe-bench/apps/frappe/frappe/app.py", line 110, in init_request
16:39:01 web.1            |     frappe.init(site=site, sites_path=_sites_path)
16:39:01 web.1            |   File "/home/giuseppe/frappe-bench/apps/frappe/frappe/__init__.py", line 138, in init
16:39:01 web.1            |     local.conf = _dict(get_site_config())
16:39:01 web.1            |   File "/home/giuseppe/frappe-bench/apps/frappe/frappe/__init__.py", line 195, in get_site_config
16:39:01 web.1            |     sys.exit(1)
16:39:01 web.1            | SystemExit: 1
16:39:01 web.1            | ----------------------------------`
